File: apps/shopping/backend/src/features/history/schemas.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ShoppingHistoryRead(BaseModel):
    id: uuid.UUID
    home_id: uuid.UUID
    name: str
    brand: str
    barcode: str | None = None
    unit: str
    purchase_count: int
    icon_tag: str | None = None
    last_purchased_at: datetime
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: apps/shopping/frontend/src/features/shopping-history/schemas.ts
import { z } from "zod";

/**
 * Zod validation schema representing a frequently purchased item search record.
 */
export const ShoppingHistorySchema = z.object({
  id: z.string().uuid(),
  home_id: z.string().uuid(),
  name: z.string().min(1),
  brand: z.string().nullable().optional(),
  barcode: z.string().nullable().optional(),
  unit: z.string().default("piece"),
  purchase_count: z.number().int().nonnegative(),
  icon_tag: z.string().nullable().optional(),
  last_purchased_at: z.string(),
  created_at: z.string(),
  updated_at: z.string(),
});
